### PR TITLE
Sign up - React query

### PR DESCRIPTION
**Pull request by Julianhm9612**
## Summary

- Added data-fetching library and configured
- Added the useMutation hook in sign up form

## Screenshots

query

## Trello Card

https://trello.com/c/4Uh52zom/45-react-query

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -7,11 +7,17 @@ import 'config/yup';
 import 'scss/application.scss';
 
 import reportWebVitals from './reportWebVitals';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+
+// Create a client
```
Comment (ebedoyaalzate) [inline src/index.tsx]: Delete this comment

File: src/screens/SignUp/index.tsx [hunk 2]
```diff
@@ -4,13 +4,14 @@ import { yupResolver } from '@hookform/resolvers/yup';
...
 import toast from 'react-hot-toast';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
 
 import Container from 'components/Container';
 import Input from 'components/Input';
 import Button from 'components/Button';
 import { MAX_PASSWORD, MIN_PASSWORD } from 'constants/constants';
 import { User } from 'types/user';
-import { useLazyRequest } from 'hooks/useRequest';
+// import { useLazyRequest } from 'hooks/useRequest';
```
Comment (ebedoyaalzate) [inline src/screens/SignUp/index.tsx]: Same here

File: src/screens/SignUp/index.tsx [hunk 3]
```diff
@@ -54,23 +55,35 @@ function SignUp() {
...
+  //     if (response) {
+  //       // Go Dashboard
+  //     }
+  //   },
+  //   withPostFailure: (err: any) => {
+  //     if (!err.errorData?.errors) {
+  //       toast.error(t('Services:genericError'));
+  //     }
+  //   }
+  // });
```
Comment (ebedoyaalzate) [inline src/screens/SignUp/index.tsx]: Same

File: src/screens/SignUp/index.tsx [hunk 4]
```diff
@@ -54,23 +55,35 @@ function SignUp() {
...
       }
     }
-  });
+  })
 
   const onSubmit = (user: User) => {
     user.locale = i18n.language;
-    request(user);
+    signUpMutation.mutate(user);
+    // request(user);
```
Comment (ebedoyaalzate) [inline src/screens/SignUp/index.tsx]: same

**Review by ebedoyaalzate — CHANGES_REQUESTED** [review]

File: src/index.tsx (hunk 1 above)
Comment (Julianhm9612) [reply src/index.tsx]: Comments deleted :suspect:

Comment (Julianhm9612) [reply src/index.tsx]: Comments deleted :suspect:

File: src/screens/SignUp/index.tsx (hunk 2 above)
Comment (Julianhm9612) [reply src/screens/SignUp/index.tsx]: Comments deleted :suspect:

File: src/screens/SignUp/index.tsx (hunk 3 above)
Comment (Julianhm9612) [reply src/screens/SignUp/index.tsx]: Comments deleted :suspect:

File: src/screens/SignUp/index.tsx (hunk 4 above)
Comment (Julianhm9612) [reply src/screens/SignUp/index.tsx]: Comments deleted :suspect:

File: src/screens/SignUp/index.tsx [hunk 5]
```diff
@@ -54,23 +54,20 @@ function SignUp() {
     resolver: yupResolver(validationSchema)
   });
 
-  const [state, loading, error, request] = useLazyRequest({
-    request: signUp,
-    withPostSuccess: (response: any) => {
-      if (response) {
-        // Go Dashboard
-      }
+  const signUpMutation = useMutation((user: User) => signUp(user), {
```
Comment (ebedoyaalzate) [inline src/screens/SignUp/index.tsx]: Use destructuring

**Review by ebedoyaalzate — APPROVED** [review]

File: src/screens/SignUp/index.tsx (hunk 5 above)
Comment (Julianhm9612) [reply src/screens/SignUp/index.tsx]: Added destructuring :suspect:

**Pull request merged**